SharperUniverse/Persistence/IDatabaseProvider.cs

```diff
@@ -0,0 +1,16 @@
+using System.Collections.Generic;
```

Comment:
Make sure this can modify specific components as well as specific entities inside a save state. You should also keep in mind it needs to be able to overwrite existing save IDs.